src/placement.c

```diff
@@ -0,0 +1,552 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <assert.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include "common/macros.h"
+#include "common/mem.h"
+#include "labwc.h"
+#include "placement.h"
+#include "ssd.h"
+#include "view.h"
+
+#define overlap_bitmap_index(bmp, i, j) \
+	(bmp)->grid[i * ((bmp)->nr_cols - 1) + j]
+
+struct overlap_bitmap {
+	int nr_rows;
+	int nr_cols;
+	int *rows;
+	int *cols;
+	int *grid;
+};
+
+static int
+compare_ints(const void *a, const void *b)
+{
+	return *(const int *)a - *(const int *)b;
+}
+
+static void
+destroy_bitmap(struct overlap_bitmap *bmp)
+{
+	assert(bmp);
+
+	if (bmp->rows) {
```

Comment:
We could save a few lines here by just using `zfree()` (labwc helper). For example (`zfree(bmp->rows)`). It does a `free()` followed by setting the variable to zero.